pkg/mco/kubeletconfig.go

```diff
@@ -0,0 +1,264 @@
+	}
+
+	builder.Definition = builder.Object
+
+	return &builder, nil
+}
+
+// Create generates a kubeletconfig in the cluster and stores the created object in struct.
+func (builder *KubeletConfigBuilder) Create() (*KubeletConfigBuilder, error) {
+	if valid, err := builder.validate(); !valid {
+		return builder, err
+	}
+
+	glog.V(100).Infof("Creating KubeletConfig %s", builder.Definition.Name)
+
+	var err error
+	if !builder.Exists() {
+		builder.Object, err = builder.apiClient.KubeletConfigs().Create(
+			context.Background(), builder.Definition, metav1.CreateOptions{})
+	}
+
+	return builder, err
+}
+
+// Delete removes the kubeletconfig.
+func (builder *KubeletConfigBuilder) Delete() error {
+	if valid, err := builder.validate(); !valid {
+		return err
+	}
+
+	glog.V(100).Infof("Deleting the kubeletconfig object %s", builder.Definition.Name)
+
+	if !builder.Exists() {
+		return fmt.Errorf("kubeletconfig cannot be deleted because it does not exist")
+	}
+
+	err := builder.apiClient.KubeletConfigs().Delete(
+		context.TODO(), builder.Object.Name, metav1.DeleteOptions{})
+
+	if err != nil {
+		return fmt.Errorf("cannot delete kubeletconfig: %w", err)
+	}
+
+	builder.Object = nil
+
+	return err
+}
+
+// Exists checks whether the given kubeletconfig exists.
+func (builder *KubeletConfigBuilder) Exists() bool {
+	if valid, _ := builder.validate(); !valid {
+		return false
+	}
+
+	glog.V(100).Infof("Checking if the kubeletconfig object %s exists", builder.Definition.Name)
+
+	var err error
+	builder.Object, err = builder.apiClient.KubeletConfigs().Get(
+		context.Background(), builder.Definition.Name, metav1.GetOptions{})
+
+	return err == nil || !k8serrors.IsNotFound(err)
+}
+
+// WithMCPoolSelector redefines kubeletconfig definition with the given machineConfigPoolSelector field.
+func (builder *KubeletConfigBuilder) WithMCPoolSelector(key, value string) *KubeletConfigBuilder {
+	if valid, _ := builder.validate(); !valid {
+		return builder
+	}
+
+	glog.V(100).Infof("Labeling the kubeletconfig %s with %s=%s", builder.Definition.Name, key, value)
+
+	if key == "" {
+		glog.V(100).Infof("The key can't be empty")
+
+		builder.errorMsg = "'key' cannot be empty"
+
+		return builder
+	}
+
+	if builder.Definition.Spec.MachineConfigPoolSelector == nil {
+		builder.Definition.Spec.MachineConfigPoolSelector = &metav1.LabelSelector{}
+	}
+
+	if builder.Definition.Spec.MachineConfigPoolSelector.MatchLabels == nil {
+		builder.Definition.Spec.MachineConfigPoolSelector.MatchLabels = map[string]string{}
+	}
+
+	builder.Definition.Spec.MachineConfigPoolSelector.MatchLabels[key] = value
+
+	return builder
+}
+
+// WithSystemReserved redefines kubeletconfig definition with the given systemreserved fields.
+func (builder *KubeletConfigBuilder) WithSystemReserved(cpu, memory string) *KubeletConfigBuilder {
+	if valid, _ := builder.validate(); !valid {
+		return builder
+	}
+
+	glog.V(100).Infof("Setting up %s cpu and %s memory to the %s kubeletconfig",
```

Comment:
nit: Based on the expected entries (i.e. cpu=200m,memory=150G) the log should be better rewritten as:
"Setting cpu=%s and memory=%s in the kubeletconfig definition"